lua/wire/client/e2descriptions.lua

```diff
@@ -804,8 +804,14 @@ E2Helper.Descriptions["hardQuota()"] = "Returns the size of the hard quota"
 E2Helper.Descriptions["perf()"] = "If used as a while loop condition, stabilizes the expression around hardquota used"
 E2Helper.Descriptions["perf(n)"] = "If used as a while loop condition, stabilizes the expression around specified number (in %)"
 E2Helper.Descriptions["entity()"] = "Gets the entity of the expression"
+E2Helper.Descriptions["getComponentName()"] = "Get the component name of the E2 itself"
+E2Helper.Descriptions["getComponentName(e:)"] = "Get the component name of another Wire entity"
+E2Helper.Descriptions["getName()"] = "Get the name of the E2 itself"
 E2Helper.Descriptions["getName(e:)"] = "Get the name of another E2"
-E2Helper.Descriptions["setName(s)"] = "Set the name of the E2"
+E2Helper.Descriptions["setComponentName(s)"] = "Set the component name of the E2 itself"
+E2Helper.Descriptions["setComponentName(e:s)"] = "Set the component name of another Wire entity"
+E2Helper.Descriptions["setName(s)"] = "Set the name of the E2 itself"
+E2Helper.Descriptions["setName(e:s)"] = "Depending on the type of entity, if it is E2 then it will set the name of another E2; otherwise, it will set component name of another Wire entity"
```

Comment:
It is to allow E2 to set its own component name... Read my previous comment as to why.